### PR TITLE
Add BSEC otp provisionning

### PR DESCRIPTION
**Pull request by GseoC**
Add a way for non-secure world to provision One Time Programmed resources using "st,non-secure-otp" device tree property.

### CODE REVIEW

File: core/drivers/stm32_bsec.c [hunk 1]
```diff
@@ -616,8 +616,41 @@ static void bsec_dt_otp_nsec_access(void *fdt, int bsec_node)
 			}
 		}
 
-		if (!fdt_getprop(fdt, bsec_subnode, "st,non-secure-otp", NULL))
+		/* Handle different kinds of non-secure accesses */
+		if (fdt_getprop(fdt, bsec_subnode,
+				"st,non-secure-otp-provisioning", NULL)) {
+			bool locked = false;
+			bool lcked_next = false;
```
Comment (etienne-lms) [inline core/drivers/stm32_bsec.c:720]: s/lcked_next/locked_next/g

File: core/drivers/stm32_bsec.c [hunk 2]
```diff
@@ -713,8 +713,41 @@ static void bsec_dt_otp_nsec_access(void *fdt, int bsec_node)
...
+
+			/* Check if write of OTP is locked */
+			if (stm32_bsec_read_permanent_lock(otp_id, &locked))
+				panic("BSEC: Couldn't read permanent lock at init");
```
Comment (etienne-lms) [inline core/drivers/stm32_bsec.c:724]: maybe use the same rtace message as below: `Cannot read permanent lock`

File: core/drivers/stm32_bsec.c [hunk 3]
```diff
@@ -713,8 +713,41 @@ static void bsec_dt_otp_nsec_access(void *fdt, int bsec_node)
...
+			 * Check if fuses of the subnode
+			 * have the same lock status
+			 */
+			for (i = 1; i < (reg_size / sizeof(uint32_t)); i++) {
+				if (stm32_bsec_read_permanent_lock(otp_id + i,
+								   &lcked_next))
+					panic("Cannot read permanent lock");
+
+				if (locked != lcked_next) {
+					EMSG("Inconsistent status OTP id %u",
```
Comment (etienne-lms) [inline core/drivers/stm32_bsec.c:736]: s/id/ID/

File: core/drivers/stm32_bsec.c [hunk 4]
```diff
@@ -713,8 +713,41 @@ static void bsec_dt_otp_nsec_access(void *fdt, int bsec_node)
...
+					     otp_id + i);
+					locked = true;
+					continue;
```
Comment (etienne-lms) [inline core/drivers/stm32_bsec.c:739]: `continue` has no effect here even if the implementation looks correct to me. I suggest to remove it.

File: core/drivers/stm32_bsec.c (hunk 1 above)
Comment (GseoC) [reply core/drivers/stm32_bsec.c:720]: Change to locked_2. I did this because of character limitation

File: core/drivers/stm32_bsec.c (hunk 2 above)
Comment (GseoC) [reply core/drivers/stm32_bsec.c:724]: Sure

File: core/drivers/stm32_bsec.c (hunk 3 above)
Comment (GseoC) [reply core/drivers/stm32_bsec.c:736]: Ack

File: core/drivers/stm32_bsec.c (hunk 4 above)
Comment (GseoC) [reply core/drivers/stm32_bsec.c:739]: Yes indeed...

**Comment by GseoC** [comment]
Hello @jforissier @etienne-lms , is there a reason QEMU check failed? I'm not finding an explicit reason in the log

**Comment by jforissier** [comment]
> Hello @jforissier @etienne-lms , is there a reason QEMU check failed? I'm not finding an explicit reason in the log

Nothing obvious indeed... the fact that we don't have the console output doesn't help. I'm restarting the job and also running it locally to see if I can reproduce the issue. In any case there is next to zero chance that the failure could be caused by this PR I think.

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by GseoC** [comment]
Force-pushed with tags applied

**Pull request merged**